src/util/endpoints.js

```diff
@@ -137,19 +139,31 @@ exports.initializeCache = async (policyOptions, config) => {
  *
  * @returns {string} - Returns the endpoint, throws error if failure occurs
  */
-exports.getEndpoint = async (switchUrl, fsp, endpointType, options = {}, renderOptions = {}) => {
+exports.getEndpoint = async (switchUrl, fsp, endpointType, options = {}, renderOptions = {}, proxyConfig = undefined) => {
   const histTimer = Metrics.getHistogram(
     'getEndpoint',
     'getEndpoint - Metrics for getEndpoint with cache hit rate',
     ['success', 'hit']
   ).startTimer()
   switchEndpoint = switchUrl
   Logger.isDebugEnabled && Logger.debug(`participantEndpointCache::getEndpoint::endpointType - ${endpointType}`)
+  let proxyId
+  const result = url => proxyConfig ? { url, proxyId } : url
   try {
     // If a service passes in `getDecoratedValue` as true, then an object
     // { value, cached, report } is returned, where value is the cached value,
     // cached is null on a cache miss.
-    const endpoints = await policy.get(fsp)
+    let endpoints = await policy.get(fsp)
+    if (!endpoints && proxyConfig) {
+      if (!proxy) {
+        const { type, ...config } = proxyConfig
+        proxy = proxyLib.createProxyCache(type, config)
+        await proxy.connect()
```

Comment:
I can not spot anywhere in the code where the `proxy` gets disconnected. Might to lead to memory leaks or issues in integration tests. You can provide a function to expose this `proxy` client so that it can be managed in the services.

Reply:
I added functions for disconnect and health check.